build-gradle/action.yml

```
@@ -226,6 +235,17 @@ runs:
         path: build/reports/problems/problems-report.html
         if-no-files-found: ignore
 
+    - name: Generate provenance attestation
+      if: |
+        ${{
+          inputs.generate-provenance == 'true' &&
+          steps.build.outputs.should-deploy == 'true' &&
+          (github.ref_name == github.event.repository.default_branch || inputs.provenance-on-pr == 'true')
+        }}
```

Generate if provenance is ON, artifact-paths is not empty

```suggestion
      if: |
          inputs.generate-provenance == 'true' && steps.build.outputs.deployed == 'true' &&
          (steps.build.outputs.artifact-paths || inputs.provenance-subject-path) &&
          (github.event_name != 'pull_request' || inputs.provenance-on-pr == 'true')
```